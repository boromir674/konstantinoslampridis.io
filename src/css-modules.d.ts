// Global type declarations for CSS modules
declare module '*.module.css' {
  const classes: { readonly [key: string]: string };
  export = classes;
}

declare module '*.module.scss' {
  const classes: { readonly [key: string]: string };
  export = classes;
}

declare module '*.module.sass' {
  const classes: { [key: string]: string };
  export default classes;
}
