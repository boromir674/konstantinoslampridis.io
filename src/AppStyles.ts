interface HeaderStyles {
  primaryColor: string;
  secondaryColor: string;
}

// Represents a set of colors corresponding to the Business Logic of the App
// Usually this interface can describe the colors of a so-called "color mode"
// For example Light/Dark mode/theme
interface Theme {
  backgroundColor: string;
  foregroundColor: string;
  buttonColor: string;
  buttonHoverColor: string;
  headerStyles: HeaderStyles;
  topHeaderPane: {
    backgroundColor: string;
  };
  navigationBar: {
    backgroundColor: string;
    textColor: string;
    hoverBackgroundColor: string;
    hoverTextColor: string;
    activatedBackgroundColor: string;
    activatedTextColor: string;
  };
  personal: {
    textColor: string;
    urlTextColor: string;
    containerBackgroundColor: string;
  };
  education: {
    title: {
      textColor: string;
      backgroundColor: string;
    };
    item: {
      backgroundColor: string;
      textColor: string;
      linkColor: string;
      onHoverBackgroundColor: string;
      onHoverTextColor: string;
    };
    containerBackgroundColor: string;
  };
  introduction: {
    containerBackgroundColor: string;
    textColor: string;
  };
  professional: {
    containerBackgroundColor: string;
    title: {
      textColor: string;
      backgroundColor: string;
    };
    item: {
      backgroundColor: string;
      textColor: string;
      linkColor: string;
      onHoverBackgroundColor: string;
      onHoverTextColor: string;
    };
  };
  footerStyles: {
    textColor: string;
    backgroundColor: string;
  };
}

// Used internally for type checking
interface ComputedTheme extends Theme {
  professional: {
    containerBackgroundColor: Theme["professional"]["containerBackgroundColor"];
    title: Theme["professional"]["title"];
    item: {
      backgroundColor: string;
      textColor: string;
      linkColor: string;
      onHoverBackgroundColor: string;
      onHoverTextColor: string;
      onHoverTransformDuration: string;
      onHoverBackgroundColorChangeDuration: string;
    };
    // defines whether there will be a delay between the the top item and the
    // bottom, on color mode switch (toggle)

    // for example if given number is 4 then from the moment the top item
    // starts to change colors (on mode switch), till the moment the bottom
    // item changed colors, 4 seconds would pass
    itemsColorModeSwitchDelay: number;
  };
  education: {
    title: {
      textColor: string;
      backgroundColor: string;
    };
    item: {
      backgroundColor: string;
      textColor: string;
      linkColor: string;
      onHoverBackgroundColor: string;
      onHoverTextColor: string;
      // Common Styling that does not depend on color
      onHoverTransformDuration: string;
      onHoverBackgroundColorChangeDuration: string;
    };
    containerBackgroundColor: string;
  };
}

// DESIGNER SPACE
// COMMON Styling Configuration between Light and Dark Modes

const commonStyling = {
  fontFamily: "Roboto, sans-serif",
  fontSize: "16px",
  fontWeight: "400",
  lineHeight: "1.5",
  letterSpacing: "0.00938em",
  // Domain Specific Styling
  professional: {
    item: {
      onHoverTransformDuration: "1.6s",
      onHoverBackgroundColorChangeDuration: "0.7s",
      // onHoverTransitionDelay: '1s',
    },
    itemsColorModeSwitchDelay: 1.2,
  },
  education: {
    item: {
      onHoverTransformDuration: "1.6s",
      onHoverBackgroundColorChangeDuration: "0.7s",
      // onHoverTransitionDelay: '1s',
    },
  },
};

const lightColorMode: Theme = {
  // <picked> color after thought
  backgroundColor: "#D6D6DF",
  foregroundColor: "#1D5556",
  // </picked>

  // foregroundColor: "#000000",
  buttonColor: "#007bff",
  buttonHoverColor: "#0056b3",
  headerStyles: {
    primaryColor: "#ffffff",
    secondaryColor: "#0056b3",
  },
  topHeaderPane: {
    backgroundColor: "#D6D6DF",
  },
  navigationBar: {
    textColor: "#000000",
    backgroundColor: "#009385",
    hoverBackgroundColor: "#fff",
    hoverTextColor: "#000000",
    activatedBackgroundColor: "#fa345f",
    activatedTextColor: "#000000",
  },
  personal: {
    textColor: "#4A4A4A",
    urlTextColor: "#2063e9",
    // containerBackgroundColor: "#f6f8fa",
    containerBackgroundColor: "#EDB6DB",
  },
  education: {
    title: {
      textColor: "#000000",
      backgroundColor: "#ffffff",
    },
    item: {
      // MY OLD DESIGN
      // backgroundColor: "#FFAD00",
      // textColor: "#333",
      onHoverBackgroundColor: "#ddd",
      onHoverTextColor: "#333",
      linkColor: "#2063e9",
      // NEW DESIGN
      textColor: "#123123",
      backgroundColor: "#ffffff",
      // textColor: "#000000",
      // onHoverBackgroundColor: "#f6f8fa",
      // onHoverTextColor: "#000000",
    },
    containerBackgroundColor: "#f6f8fa",
  },
  introduction: {
    containerBackgroundColor: "#ffffff",
    textColor: "#000000",
  },
  professional: {
    title: {
      textColor: "#000000",
      backgroundColor: "#ffffff",
    },
    item: {
      textColor: "#000000",
      backgroundColor: "#ffffff",
      linkColor: "#1651f3",
      onHoverBackgroundColor: "#a4e2c7",
      onHoverTextColor: "#000000",
    },
    containerBackgroundColor: "#f6f8fa",
  },
  footerStyles: {
    textColor: "#1D5556",
    backgroundColor: "#D6D6DF",
  },
};

const darkColorMode: Theme = {
  // <picked> color after thought
  backgroundColor: "#D6D6DF",
  foregroundColor: "#1D5556",
  // </picked>
  // backgroundColor: "#333333",
  // foregroundColor: "#ffffff",

  buttonColor: "#007bff",
  buttonHoverColor: "#0096ff",
  headerStyles: {
    primaryColor: "#333333",
    secondaryColor: "#0096ff",
  },
  topHeaderPane: {
    backgroundColor: "#463507",
  },
  navigationBar: {
    backgroundColor: "#333333",
    textColor: "#ffffff",
    hoverBackgroundColor: "#fff",
    hoverTextColor: "#ffffff",
    activatedBackgroundColor: "#ff345f",
    activatedTextColor: "#ffffff",
  },
  // PERSONAL INFO / CONTACT / LINKS
  personal: {
    textColor: "#EDEDED",
    urlTextColor: "#D98C66",
    // containerBackgroundColor: "#222222",
    containerBackgroundColor: "#2D0B22",
  },
  // EDUCATION SECTION
  education: {
    title: {
      textColor: "#ffffff",
      backgroundColor: "#333333",
    },
    item: {
      backgroundColor: "#333333",
      textColor: "#ffffff",
      linkColor: "#D98C66",
      onHoverBackgroundColor: "#444444",
      onHoverTextColor: "#ffffff",
    },
    containerBackgroundColor: "#222222",
  },
  introduction: {
    containerBackgroundColor: "#201f1f",
    textColor: "#d7e5db",
  },
  professional: {
    title: {
      textColor: "#d7e5db",
      backgroundColor: "#201f1f",
    },
    item: {
      textColor: "#d7e5db",
      backgroundColor: "#201f1f",
      linkColor: "#aec0f1",
      onHoverBackgroundColor: "#494343",
      onHoverTextColor: "#d7e5db",
    },
    containerBackgroundColor: "#1d2421",
  },
  footerStyles: {
    textColor: "#D6D6DF",
    backgroundColor: "#1D5556",
  },
};

// Computing the final (Light + Common Styles, Dark + Common Styles) Theme
type CommonStyling = typeof commonStyling;

const mergeStylings = (theme: Theme, commonStyling: CommonStyling) => {
  return {
    ...theme,
    ...commonStyling,

    professional: {
      ...theme.professional,
      ...commonStyling.professional,
      item: {
        ...theme.professional.item,
        ...commonStyling.professional.item,
      },
    },

    education: {
      ...theme.education,
      ...commonStyling.education,
      item: {
        ...theme.education.item,
        ...commonStyling.education.item,
      },
    },
  };
};

const lightTheme: ComputedTheme = mergeStylings(lightColorMode, commonStyling);
const darkTheme: ComputedTheme = mergeStylings(darkColorMode, commonStyling);

export { ComputedTheme, lightTheme, darkTheme };
