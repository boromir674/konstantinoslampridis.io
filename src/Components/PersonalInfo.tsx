import React from "react";
import styled from "@emotion/styled";
import PersonalInfoInterface from "../PersonalInfoInterface";

interface ContainerProps {
  theme: {
    containerBackgroundColor: string;
    textColor: string;
  };
}
interface URLProps {
  theme: {
    textColor: string;
  };
}
interface PersonalInfoProps {
  userData: PersonalInfoInterface;
  theme: {
    containerBackgroundColor: string;
    textColor: string;
    linkColor: string;
  };
}

const Container = styled.div<ContainerProps>`
  display: flex;
  flex-wrap: wrap;
  // background-color: #f6f8fa;
  // color: #333;
  padding: 20px;
  border-radius: 10px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  background-color: ${(props) => props.theme.containerBackgroundColor};
  color: ${(props) => props.theme.textColor};
`;

const InfoItem = styled.div`
  flex: 1 1 50%;
  padding: 10px;
`;

const InfoKey = styled.span`
  font-weight: bold;
  font-size: 18px;
`;

const InfoValue = styled.span`
  display: inline-flex;
  align-items: center;
  font-size: 16px;
  margin-top: 5px;
`;

const Link = styled.a<URLProps>`
  color: ${(props) => props.theme.textColor};
  // color: #2063e9;
  // text-decoration: none;
  transition: color 0.3s ease;

  &:hover {
    color: #ff7f50;
  }
`;

const MyLink = styled(Link)`
  margin-right: 10px;
`;

// const GitLabLink = styled(Link)`
//   margin-right: 10px;
// `;

// const LinkedInLink = styled(Link)`
//   margin-right: 10px;
// `;

const PersonalInfo: React.FC<PersonalInfoProps> = ({
  userData: { name, email, github, gitlab, linkedin },
  theme,
}) => {
  return (
    <Container theme={theme}>
      <InfoItem>
        <InfoKey>Name:</InfoKey> <InfoValue>{name}</InfoValue>
      </InfoItem>
      <InfoItem>
        <InfoKey>Email:</InfoKey> <InfoValue>{email}</InfoValue>
      </InfoItem>
      <InfoItem>
        <InfoKey>Github:</InfoKey>{" "}
        <InfoValue>
          <MyLink
            href={`https://github.com/${github}`}
            target="_blank"
            rel="noopener noreferrer"
            theme={{
              textColor: theme.linkColor,
            }}
          >
            {github}
          </MyLink>
        </InfoValue>
      </InfoItem>
      <InfoItem>
        <InfoKey>Gitlab:</InfoKey>{" "}
        <InfoValue>
          <MyLink
            href={`https://gitlab.com/${gitlab}`}
            target="_blank"
            rel="noopener noreferrer"
            theme={{
              textColor: theme.linkColor,
            }}
          >
            {gitlab}
          </MyLink>
        </InfoValue>
      </InfoItem>
      <InfoItem>
        <InfoKey>LinkedIn:</InfoKey>{" "}
        <InfoValue>
          <MyLink
            href={`https://${linkedin}`}
            target="_blank"
            rel="noopener noreferrer"
            theme={{
              textColor: theme.linkColor,
            }}
          >
            {linkedin}
          </MyLink>
        </InfoValue>
      </InfoItem>
    </Container>
  );
};

export default PersonalInfo;
