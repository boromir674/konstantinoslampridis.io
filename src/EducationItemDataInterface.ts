/** 
* A Person's Education Item information.
* @summary This is how the App represents a Person's Education item.
* @interface EducationItemData
* @property {string} degree_title - The title of the degree.
* @property {string} university_name - The name of the university.
* @property {string} location - The location of the university.
* @property {string} duration - The duration of the degree/studies (ie '2018', '2014 - 2019').
* @property {string} thesis_title - The title of the thesis.
* @property {string[]} topics - List of courses/topics in the degree curriculum.
*/
interface EducationItemData {
    degree_title: string;
    university_name: string;
    location: string;
    duration: string;
    thesis_title: string;
    topics: string[];
  }

export default EducationItemData;
