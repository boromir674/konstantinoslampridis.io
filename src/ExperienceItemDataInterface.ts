/**
 * A Person's Experience Item information.
 * @summary This is how the App represents a Person's Experience item.
 * @interface ExperienceItemData
 * @property {string} title - The Job title of the Person.
 * @property {string} company - The Company name
 * @property {string} location - The location of the company (or Remote)
 * @property {string} duration - The duration of the occupation (ie '2020 - 2021').
 * @property {string} description - The description of the job.
 * @property {string[]} activities - List of activities performed in the job.
 * @property {string[]} technology_tags - List of technologies used in the job.
 */
interface ExperienceItemData {
  title: string;
  company: string
  location: string;
  duration: string;
  description: string;
  activities: string[];
  technology_tags: string[];
}

export default ExperienceItemData;
