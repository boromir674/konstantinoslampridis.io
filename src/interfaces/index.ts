import type { LayoutInterface, LayoutsObject } from './GridLayoutInterfaces';
import IconStyles from './SVGIconStyles';

export type { IconStyles, LayoutInterface, LayoutsObject };
