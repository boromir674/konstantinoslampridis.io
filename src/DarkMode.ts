/* DESIGN your Dark Mode Color Theme */
import DESIGN_TOKENS from "./design-system/tokens.json";
import { Theme } from "./AppStyles";

/////////// DARK MODE ///////////

const darkColorMode: Theme = {
  // This affects some out most backgound color
  backgroundColor: DESIGN_TOKENS["--md-sys-color-background-dark"],

  //// TOP HEADER PANE ////
  topHeaderPane: {
    backgroundColor: DESIGN_TOKENS["--md-sys-color-background-dark"],
  },
  //// NAVIGATION BAR in big-screen page view ////
  navigationBar: {
    backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
    textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
    hoverBackgroundColor: DESIGN_TOKENS["--md-sys-color-primary-dark"],
    hoverTextColor: DESIGN_TOKENS["--md-sys-color-on-primary-dark"],
    activatedBackgroundColor: DESIGN_TOKENS["--md-sys-color-secondary-dark"],
    activatedTextColor: DESIGN_TOKENS["--md-sys-color-on-secondary-dark"],
  },
  //// THEME SWITCH ////
  themeSwitch: {
    backgroundColor: DESIGN_TOKENS["--md-sys-color-secondary-dark"],
    handleBackgroundColor: DESIGN_TOKENS["--md-sys-color-on-secondary-dark"],
    backgroundColorActive: DESIGN_TOKENS["--md-sys-color-on-secondary-dark"],
    handleBackgroundColorActive: DESIGN_TOKENS["--md-sys-color-surface-dark"],
  },
  ///// INTRODUCTION /////
  introduction: {
    // containerBackgroundColor: DESIGN_TOKENS['--md-sys-color-primary-container-dark'],
    containerBackgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
    textColor: DESIGN_TOKENS["--md-sys-color-on-primary-container-dark"],
  },
  // PERSONAL INFO / CONTACT / LINKS
  personal: {
    containerBackgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
    textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
    urlTextColor: DESIGN_TOKENS["--md-sys-color-on-surface-variant-dark"],
    externalURLSVGColor:
      DESIGN_TOKENS["--md-sys-color-on-surface-variant-dark"],
  },
  //////// EDUCATION SECTION DARK ////////
  education: {
    // title: {
    //   textColor: "#ffffff",
    //   backgroundColor: "#333333",
    // },
    item: {
      backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
      textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
      linkColor: "#2063e9",
      onHoverBackgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
      onHoverTextColor:
        DESIGN_TOKENS["--md-sys-color-on-tertiary-container-dark"],
      tag: {
        textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
        backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
        onHoverBackgroundColor: DESIGN_TOKENS["--md-sys-color-tertiary-dark"],
        onHoverTextColor: DESIGN_TOKENS["--md-sys-color-on-tertiary-dark"],
        outlineColor: DESIGN_TOKENS["--md-sys-color-outline-dark"],
      },
    },
    containerBackgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
  },

  ///// PROFESSIONAL DARK /////
  professional: {
    // governs color of spaces around title
    containerBackgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
    title: {
      backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
      textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
    },
    item: {
      backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
      textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
      linkColor: "#2063e9",
      onHoverBackgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
      onHoverTextColor:
        DESIGN_TOKENS["--md-sys-color-on-tertiary-container-dark"],
      tag: {
        backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
        textColor: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
        onHoverBackgroundColor: DESIGN_TOKENS["--md-sys-color-primary-dark"],
        onHoverTextColor: DESIGN_TOKENS["--md-sys-color-on-primary-dark"],
        outlineColor: DESIGN_TOKENS["--md-sys-color-outline-dark"],
      },
    },
  },
  // PORTFOLIO DARK
  portfolio: {
    container: {
      backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
    },
    sectionHeader: {
      backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
      color: DESIGN_TOKENS["--md-sys-color-on-secondary-container-dark"],
    },
    // Portfolio PROJECT ITEM - DARK
    item: {
      // TODO; fix bug: item color backgroundColor has no effect
      backgroundColor: DESIGN_TOKENS["--md-sys-color-secondary-container-dark"],
      color: DESIGN_TOKENS["--md-sys-color-on-secondary-container-dark"],
      outline: {
        color: DESIGN_TOKENS["--md-sys-color-outline-dark"],
      },
      // RESOURCE LINKS - PANE - DARK
      resourceLinks: {
        backgroundColor: DESIGN_TOKENS['--md-sys-color-secondary-container-dark'],
        headerColor: DESIGN_TOKENS['--md-sys-color-on-secondary-container-dark'],
        item: {
          backgroundColor: DESIGN_TOKENS['--md-sys-color-secondary-container-dark'],
          color: DESIGN_TOKENS['--md-sys-color-on-secondary-container-dark'],
          onHoverBackgroundColor: DESIGN_TOKENS['--md-sys-color-tertiary-container-dark'],
          onHoverColor: DESIGN_TOKENS['--md-sys-color-on-tertiary-container-dark'],
        }
      },
      // RELEASES PANE - DARK
      releases: {
        backgroundColor: DESIGN_TOKENS["--md-sys-color-surface-dark"],
        color: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
        // RELEASE ITEM
        item: {
          backgroundColor: DESIGN_TOKENS['--md-sys-color-surface-variant-dark'],
          color: DESIGN_TOKENS["--md-sys-color-on-surface-dark"],
          onHoverBackgroundColor: DESIGN_TOKENS["--md-sys-color-tertiary-dark"],
          onHoverColor: DESIGN_TOKENS["--md-sys-color-on-tertiary-dark"],
        },
      },
    },
  },
  footerStyles: {
    textColor: "#D6D6DF",
    backgroundColor: "#1D5556",
  },
};

export default darkColorMode;
