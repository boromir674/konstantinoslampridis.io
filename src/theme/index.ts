// Theme / Style initialization Operations
import lightMode from "../LightMode";
import darkMode from "../DarkMode";
import commonStyles from "./CommonTheme";

import { type ComputedTheme, type Theme } from "./AppStyles";


// Computing the final (Light + Common Styles, Dark + Common Styles) Theme
type CommonStyling = typeof commonStyles;

const mergeStylings = (
  theme: Theme,
  commonStyling: CommonStyling
): ComputedTheme => {
  return {
    ...theme,
    ...commonStyling,

    navigationBar: {
      ...theme.navigationBar,
      ...commonStyling.navigationBar
    },

    personal: {
      ...theme.personal,
      ...commonStyling.personal,
    },

    professional: {
      containerBackgroundColor: theme.professional.containerBackgroundColor,
      title: {
        ...theme.professional.title,
        ...commonStyling.professional.title,
      },
      itemsColorModeSwitchDelay: commonStyling.professional.itemsColorModeSwitchDelay,
      item: {
        ...theme.professional.item,
        ...commonStyling.professional.item,
        tag: {
          ...theme.professional.item.tag,
          ...commonStyling.professional.item.tag,
        },
      },
    },

    education: {
      ...theme.education,
      ...commonStyling.education,
      // title: {
      //   ...theme.education.title,
      //   ...commonStyling.education.title,
      // },
      item: {
        ...theme.education.item,
        padding: commonStyling.education.item.padding,
        onHoverTransformDuration: commonStyling.education.item.onHoverTransformDuration,
        onHoverBackgroundColorChangeDuration: commonStyling.education.item.onHoverBackgroundColorChangeDuration,
        tag: {
          ...theme.education.item.tag,
          ...commonStyling.education.item.tags,
        },
        body: commonStyling.education.item.body,
        degreeTitle: commonStyling.education.item.degreeTitle,
      },
    },
    portfolio: {
      ...theme.portfolio,
      ...commonStyling.portfolio,
      item: {
        ...theme.portfolio.item,
        ...commonStyling.portfolio.item,
        outline: {
          ...theme.portfolio.item.outline,
          ...commonStyling.portfolio.item.outline,
        },
        // Project Links - Pane
        resourceLinks: {
          ...theme.portfolio.item.resourceLinks,
          item: {
            ...theme.portfolio.item.resourceLinks.item,
            ...commonStyling.portfolio.item.resourceLinks.item,
          },
          header: {
            fontFamily: "",
            fontSize: ""
          }
        },
        // Software Releases - Pane
        releases: {
          ...theme.portfolio.item.releases,
          ...commonStyling.portfolio.item.releases,
          headerFontFamily: commonStyling.portfolio.item.releases.headerFont,
          fontFamily: commonStyling.portfolio.item.releases.headerFont,
          headerMarginBottom: commonStyling.portfolio.item.releases.headerMarginBottom,
          item: {
            ...theme.portfolio.item.releases.item,
            ...commonStyling.portfolio.item.releases.item,
            fontFamily: commonStyling.portfolio.item.releases.item.font,
          },
        },
      },
    },
    footerStyles: {
      ...theme.footerStyles,
      ...commonStyling.footerStyles,
      svgStyles: {
        ...theme.footerStyles.svgStyles,
        ...commonStyling.footerStyles.svgStyles,
      }
    },
  };
};


const lightTheme: ComputedTheme = mergeStylings(lightMode, commonStyles);
const darkTheme: ComputedTheme = mergeStylings(darkMode, commonStyles);

export { lightTheme, darkTheme };
