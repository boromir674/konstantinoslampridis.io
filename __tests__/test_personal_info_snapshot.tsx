/**
* @jest-environment jsdom
*/

import React from "react";
// Helper for rendering in customizable viewports (ie custom width)
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";

// Business-logic Imports
import PersonalInfo, { PersonalInfoProps } from "../src/Components/PersonalInfo";
import { lightTheme } from '../src/theme';


type LinkData = {
    name: string;
    id: string;
    url: string;
};

describe("Test Personal Section with width = 700", () => {

    // GIVEN an Array of Links
    const links: LinkData[] = [
        {
            name: "github",
            id: "github",
            url: "test-github-url"
        },
        {
            name: "gitlab",
            id: "gitlab",
            url: "test-gitlab-url"
        },
        {
            name: "linkedin",
            id: "linkedin",
            url: "test-linkedin-url"
        },
    ]

    const name2Url = links.reduce(
        (acc: any, { name, id, url }: any) => ({ ...acc, [id]: url }),
        {}
    );

    // PROPS VALUES
    const args: PersonalInfoProps = {
        // same interface as the props of the Component
        userData: {
            name: 'test-name',
            email: 'test-email',
            // github: name2Url["github"],
            github: links[0].url,
            gitlab: name2Url["gitlab"],
            linkedin: name2Url["linkedin"],
        },
        theme: {
            ...lightTheme.personal,
            linkColor: lightTheme.personal.urlTextColor,
        }

    };

    test("matches the snapshot", () => {
        // WHEN the Component is rendered in a Test DOM
        const { container: tree } = render(
            <ResponsiveContext.Provider value={{ width: 700 }}>
                <PersonalInfo {...args} />
            </ResponsiveContext.Provider>
        );  
        expect(tree).toMatchSnapshot();
        // AND there should be link with readable text 'test-github-url'
        // expect(tree).toHaveTextContent("test-linkedin-url");
    });
    
});
