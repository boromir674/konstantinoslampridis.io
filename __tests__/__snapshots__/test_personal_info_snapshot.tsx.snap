// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Personal Section with width = 700 matches the snapshot 1`] = `
<div>
  <div
    class="css-7xsp98"
  >
    <div
      class="css-kp9dvk"
    >
      <span
        class="css-bwb5ro"
      >
        Name:
      </span>
       
      <span
        class="css-1pgu2xd"
      >
        test-name
      </span>
    </div>
    <div
      class="css-kp9dvk"
    >
      <span
        class="css-bwb5ro"
      >
        Email:
      </span>
       
      <span
        class="css-1pgu2xd"
      >
        test-email
      </span>
    </div>
    <div
      class="css-kp9dvk"
    >
      <span
        class="css-bwb5ro"
      >
        Github:
      </span>
       
      <span
        class="css-1pgu2xd"
      >
        <a
          class="css-1gyz2k7"
          href="https://test-github-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          test-github-url
           
          <svg
            fill="none"
            height="10px"
            stroke="#534341"
            stroke-width="5.108"
            viewBox="-2.56 -2.56 69.12 69.12"
            width="10px"
          >
            <g
              id="SVGRepo_iconCarrier"
            >
              <path
                d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
              />
              <polyline
                points="40.32 8.6 55.4 8.6 55.4 24.18"
              />
              <line
                x1="19.32"
                x2="54.61"
                y1="45.72"
                y2="8.91"
              />
            </g>
          </svg>
        </a>
      </span>
    </div>
    <div
      class="css-kp9dvk"
    >
      <span
        class="css-bwb5ro"
      >
        Gitlab:
      </span>
       
      <span
        class="css-1pgu2xd"
      >
        <a
          class="css-1gyz2k7"
          href="https://test-gitlab-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          test-gitlab-url
           
          <svg
            fill="none"
            height="10px"
            stroke="#534341"
            stroke-width="5.108"
            viewBox="-2.56 -2.56 69.12 69.12"
            width="10px"
          >
            <g
              id="SVGRepo_iconCarrier"
            >
              <path
                d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
              />
              <polyline
                points="40.32 8.6 55.4 8.6 55.4 24.18"
              />
              <line
                x1="19.32"
                x2="54.61"
                y1="45.72"
                y2="8.91"
              />
            </g>
          </svg>
        </a>
      </span>
    </div>
    <div
      class="css-kp9dvk"
    >
      <span
        class="css-bwb5ro"
      >
        LinkedIn:
      </span>
       
      <span
        class="css-1pgu2xd"
      >
        <a
          class="css-1gyz2k7"
          href="https://test-linkedin-url"
          rel="noopener noreferrer"
          target="_blank"
        >
          test-linkedin-url
           
          <svg
            fill="none"
            height="10px"
            stroke="#534341"
            stroke-width="5.108"
            viewBox="-2.56 -2.56 69.12 69.12"
            width="10px"
          >
            <g
              id="SVGRepo_iconCarrier"
            >
              <path
                d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
              />
              <polyline
                points="40.32 8.6 55.4 8.6 55.4 24.18"
              />
              <line
                x1="19.32"
                x2="54.61"
                y1="45.72"
                y2="8.91"
              />
            </g>
          </svg>
        </a>
      </span>
    </div>
    <div
      class="css-kp9dvk"
    >
       
      <span
        class="css-bwb5ro"
      >
        Curricular Docs:
      </span>
       
      <div
        class="css-5l6624"
      >
        <div
          class="css-1rstcrp"
        >
          <a
            class="css-1gyz2k7"
            href="https://konstantinos-lampridis-cv-documents.s3.eu-central-1.amazonaws.com/main_resume.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            '1 Page Resume' 
            <svg
              fill="none"
              height="10px"
              stroke="#534341"
              stroke-width="5.108"
              viewBox="-2.56 -2.56 69.12 69.12"
              width="10px"
            >
              <g
                id="SVGRepo_iconCarrier"
              >
                <path
                  d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                />
                <polyline
                  points="40.32 8.6 55.4 8.6 55.4 24.18"
                />
                <line
                  x1="19.32"
                  x2="54.61"
                  y1="45.72"
                  y2="8.91"
                />
              </g>
            </svg>
          </a>
        </div>
        <div
          class="css-1rstcrp"
        >
          <a
            class="css-1gyz2k7"
            href="https://konstantinos-lampridis-cv-documents.s3.eu-central-1.amazonaws.com/main_cv%2Bprojects.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            'CV + Projects' 
            <svg
              fill="none"
              height="10px"
              stroke="#534341"
              stroke-width="5.108"
              viewBox="-2.56 -2.56 69.12 69.12"
              width="10px"
            >
              <g
                id="SVGRepo_iconCarrier"
              >
                <path
                  d="M55.4,32V53.58a1.81,1.81,0,0,1-1.82,1.82H10.42A1.81,1.81,0,0,1,8.6,53.58V10.42A1.81,1.81,0,0,1,10.42,8.6H32"
                />
                <polyline
                  points="40.32 8.6 55.4 8.6 55.4 24.18"
                />
                <line
                  x1="19.32"
                  x2="54.61"
                  y1="45.72"
                  y2="8.91"
                />
              </g>
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
